### DISCUSSION TITLE
What are graphics5 and audio2?

### DISCUSSION THREAD

**Discussion by algorev** [Kode/Kha#1336] [Q&A] [answered]
I'm new here. I've looked at the wiki and the API docs, and i've got two questions:

1. What is audio2? It seems to be a more complex audio API than audio1, but i have no idea what it does.
2. What is graphics5? It appears in the API docs, but not on the wiki. Is it the in-progress implementation of graphics3? Is it something else entirely? Is it usable?

**Comment by RobDangerous** [comment, marked as answer]
1. Audio2 is an API where you just send a stream of audio-samples - see the Audio2-based Audio1 implementation (kha.audio2.Audio1) for an example of how to use it.
2. Graphics5 is a lower level GPU API where you have to synchronize resources of GPU resources manually and which allows multithreaded usage. But it's only a Kinc thing, not a Kha thing - I deactivated it in Kha because I consider it to be too advanced for Haxe programmers. The code is still in there, that's why you can still see a graphics5_ package. If Haxe will ever receive better multithreading support I might revive it but I don't think that will happen.

**Reply by algorev** [reply]
I see. Thanks for the info!
